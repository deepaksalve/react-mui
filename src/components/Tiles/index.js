import React from "react";
import Grid from "@mui/material/Grid";

export default function Tiles({ items, renderItem }) {
  return (
    <Grid container spacing={2} sx={{ px: 2 }} data-testid="tiles">
      {Array.isArray(items) &&
        items.map((item) => (
          <Grid
            item
            xs={6}
            sm={4}
            md={3}
            lg={3}
            xl={2}
            key={item.uuid}
            data-testid="tile"
          >
            {renderItem(item)}
          </Grid>
        ))}
    </Grid>
  );
}
