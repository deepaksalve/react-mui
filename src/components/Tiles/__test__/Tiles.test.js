import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import Tiles from "../";

const tiles = [
  {
    uuid: 1,
    title: "Tile 1",
  },
  {
    uuid: 2,
    title: "Tile 2",
  },
  {
    uuid: 3,
    title: "Tile 3",
  },
  {
    uuid: 4,
    title: "Tile 4",
  },
];
describe("Headings", () => {
  describe("Title", () => {
    test("renders", () => {
      render(<Tiles renderItem={() => false} />);

      const elm = screen.getByTestId("tiles");

      expect(elm).toBeInTheDocument();
    });

    test("should have exact tiles", () => {
      render(<Tiles items={tiles} renderItem={() => false} />);

      const elm = screen.getAllByTestId("tile");

      expect(elm.length).toEqual(tiles.length);
    });

    test("matches snapshot - empty", () => {
      const tree = renderer.create(<Tiles renderItem={() => false} />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test("matches snapshot - 4 items", () => {
      const tree = renderer
        .create(<Tiles items={tiles} renderItem={() => false} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
