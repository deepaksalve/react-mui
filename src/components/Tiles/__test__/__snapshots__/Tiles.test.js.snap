// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headings Title matches snapshot - 4 items 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-3988i3-MuiGrid-root"
  data-testid="tiles"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-2 css-1cxeybu-MuiGrid-root"
    data-testid="tile"
  />
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-2 css-1cxeybu-MuiGrid-root"
    data-testid="tile"
  />
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-2 css-1cxeybu-MuiGrid-root"
    data-testid="tile"
  />
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-2 css-1cxeybu-MuiGrid-root"
    data-testid="tile"
  />
</div>
`;

exports[`Headings Title matches snapshot - empty 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-3988i3-MuiGrid-root"
  data-testid="tiles"
/>
`;
